trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/credentials/file/TestFileBasedCredentialsProvider.java

```diff
@@ -34,15 +38,17 @@ public static void setUpClass()
     {
         File configFile = new File(Resources.getResource("credentials.json").getFile());
         FileBasedCredentialsProviderConfig config = new FileBasedCredentialsProviderConfig().setCredentialsFile(configFile);
-        credentialsProvider = new FileBasedCredentialsProvider(config, new ObjectMapperProvider().get());
+        ObjectMapper objectMapper = new ObjectMapperProvider().get();
+        objectMapper.registerModule(new SimpleModule().addAbstractTypeMapping(Identity.class, TestingIdentity.class));
```

Comment:
Why is this needed?

Reply:
Since Identity is an interface, we would have to let know the object mapper which implementation of Identity should be deserialised

Reply:
You can always inject additional Guice Modules via:

```
Multibinder.newSetBinder(binder, Module.class).addBinding().to(...)
```